service_test.go

```diff
@@ -91,10 +104,57 @@ func Test_GivenDeviceId_ServiceReturnsDeviceObject(t *testing.T) {
 	assert.Equal(t, &Device{Name: "name"}, dev)
 }
 
-func Test_GivenIdThatDoesntExist_ServiceReturnsErrorNotfound(t *testing.T) {
-	out := NewService(&mockDao{returnErr: ErrNotFound("")})
+func Test_GetDevice_GivenIdThatDoesntExist_ServiceReturnsNil(t *testing.T) {
+	out := NewService(&mockDao{returnErr: nil})
 
 	_, err := out.GetDevice(1)
 
-	assert.Error(t, err)
+	assert.NoError(t, err)
+}
+
+func Test_GetManyDevices_GivenList_ServiceReturnsList(t *testing.T) {
+	out := NewService(&mockDao{data: []Device{{Name: "test name"}}})
+
+	devs, err := out.GetManyDevices(0, 0)
+
+	expected := []Device{{Name: "test name"}}
+
+	assert.NoError(t, err)
+	assert.Equal(t, expected, devs)
+}
+
+func Test_GetManyDevices_GivenDaoError_ServiceReturnsError(t *testing.T) {
+	out := NewService(&mockDao{returnErr: ErrDao("")})
+
+	_, err := out.GetManyDevices(1, 0)
+
+	assert.Equal(t, ErrDao(""), err)
+}
+
+func Test_GetManyDevices_GivenLimitAndPage_ServiceReturnsCorrectAmmountOfDevices(t *testing.T) {
+	t.Skip()
+	var mockData []Device
+	for i := 0; i < 5; i++ {
+		mockData = append(mockData, Device{Name: "test"})
+	}
+
+	out := NewService(&mockDao{data: mockData})
+
+	tests := map[string]struct {
+		expected    []Device
+		limit, page int
+	}{
+		"zero limit":        {expected: mockData, limit: 0, page: 0},
+		"limit 2 page 1":    {expected: mockData[:2], limit: 2, page: 1},
+		"limit 2 page 2":    {expected: mockData[:2], limit: 2, page: 2},
+		"limit 4 page 1":    {expected: mockData[:1], limit: 4, page: 1},
+		"zero limit page 3": {expected: []Device{}, limit: 0, page: 3},
+	}
+
+	for name, tc := range tests {
```

Comment:
Regarding the table tests, try such a change:
```suggestion
	for name, tc := range tests {
	   t.Run(name, func(t *testing.T) {
	       actual, err := out.GetManyDevices(tc.limit, tc.page)

		assert.NoError(t, err, name)
		assert.Equal(t, tc.expected, actual, name)
	  })
	}
```

This will give you more detailed results